components/layout_thread/lib.rs

```diff
@@ -898,6 +898,7 @@ impl LayoutThread {
                 Msg::ExitNow => {
                     debug!("layout thread is exiting...");
                     self.exit_now();
+                    self.background_hang_monitor.unregister();
```

Comment:
Ok, looks good! I think the last change required is moving this call to inside `exit_now`(see a earlier comment on that)